crates/oct-cloud/src/aws.rs

```diff
@@ -314,6 +341,58 @@ impl Ec2Instance {
             instance_profile: Some(instance_profile),
         }
     }
+    pub fn to_state(&self) -> Ec2InstanceState {
+        Ec2InstanceState {
+            id: self.id.clone().unwrap_or_default(),
+            arn: self.arn.clone().unwrap_or_default(),
+            public_ip: self.public_ip.clone().unwrap_or_default(),
+            public_dns: self.public_dns.clone().unwrap_or_default(),
+            region: self.region.clone(),
+            ami: self.ami.clone(),
+            instance_type: self.instance_type.clone().to_string(),
+            name: self.name.clone(),
+            instance_profile: self
+                .instance_profile
+                .as_ref()
+                .map(|profile| profile.to_state()),
+        }
+    }
+    pub async fn new_from_state(
+        state: Ec2InstanceState,
+    ) -> Result<Self, Box<dyn std::error::Error>> {
+        // Load AWS configuration
+        let region_provider = aws_sdk_ec2::config::Region::new(state.region.clone());
+        let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+            .region(region_provider)
+            .load()
+            .await;
+
+        let ec2_client = aws_sdk_ec2::Client::new(&config);
+        let iam_client = aws_sdk_iam::Client::new(&config);
```

Comment:
`iam_client` should be created in `InstanceProfile::new_from_state` and `InstanceRole::new_from_state`. The same logic as in `new` methods